lib/kino_vega_lite/chart_cell.ex

```diff
@@ -128,6 +129,16 @@ defmodule KinoVegaLite.ChartCell do
     {:noreply, ctx}
   end
 
+  def handle_event("update_field", %{"field" => field, "value" => value, "layer" => idx}, ctx)
+      when field in @fields_with_types do
+    {updated_fields, updated_layer} = updates_for_field_with_types(ctx, field, idx, value)
+    updated_layers = List.replace_at(ctx.assigns.layers, idx, updated_layer)
+    ctx = update_in(ctx.assigns, fn assigns -> Map.put(assigns, :layers, updated_layers) end)
```

Comment:
```suggestion
    ctx = assign(ctx, layers: updated_layers)
```

We do this in several places, so we should update all of them :)